rpc/rpc_client.go

```diff
@@ -294,8 +298,10 @@ func Connect(connConfig Config, suppressRegister bool, dispatcherFuncs map[strin
 		conn.Write([]byte("proto2"))
 		conn.Write([]byte{byte(len(connID))})
 		conn.Write([]byte(connID))
+		connectionDialingWG.Done()
```

Comment:
let's add a `defer` to this line, and move it to the start of the function, just in case it doesn't reach here